### PR TITLE
Feature/additional askbot events

### PR DESCRIPTION
**Pull request by vxsx**
- some additional askbot event triggers for customization purposes
- fix behaviour of convert comment to aswer buttons
- fix behaviour of vote buttons on comments

### CODE REVIEW

File: askbot/media/js/post.js [hunk 1]
```diff
@@ -2479,13 +2488,15 @@ Comment.prototype.setContent = function (data) {
     }
 
     if (this._is_convertible) {
-        if (this._convert_link !== undefined) {
-            this._convert_link.dispose();
```
Comment (vxsx) [inline askbot/media/js/post.js]: so here dispose basically was removing the form from DOM, and on line 2493 we were trying to replace the contents of that form with correct data which wasn't very helpful

File: askbot/media/js/post.js [hunk 2]
```diff
@@ -2199,7 +2206,7 @@ EditCommentForm.prototype.getSuppressEmail = function () {
 };
 
 EditCommentForm.prototype.setSuppressEmail = function (bool) {
-    this._element.find('input[name="suppress_email"]').prop('checked', bool);
+    this._element.find('input[name="suppress_email"]').prop('checked', bool).trigger('change');
```
Comment (vxsx) [inline askbot/media/js/post.js]: if there's a custom checkbox solution - just changing the prop won't suffice

File: askbot/media/js/post.js [hunk 3]
```diff
@@ -2153,6 +2159,7 @@ EditCommentForm.prototype.createDom = function () {
         var checkBox = this.makeElement('input');
         checkBox.attr('type', 'checkbox');
         checkBox.attr('name', 'suppress_email');
+        checkBox.attr('id', 'suppress_email');
```
Comment (vxsx) [inline askbot/media/js/post.js]: without id input and label were not connected so click on label didn't change input

File: askbot/media/js/post.js [hunk 4]
```diff
@@ -576,8 +576,10 @@ CommentVoteButton.prototype.setScore = function (score) {
  */
 CommentVoteButton.prototype.setVoted = function (voted) {
```
Comment (vxsx) [inline askbot/media/js/post.js]: so in `Comment.prototype.setContent` there's a line `vote.setVoted(data.upvoted_by_user)` which means that `voted` argument can be false, but previous code always set `upvoted` class no matter the value, which doesn't seem correct

File: askbot/media/js/post.js (hunk 2 above)
Comment (evgenyfadeev) [inline askbot/media/js/post.js]: ok. It worked because the property is read just before making the submit.
Nothing was listening to the change event.

**Pull request merged**